cli/smartcontract/generate.go

```diff
@@ -0,0 +1,155 @@
+	callflag.AllowCall,
+	callflag.AllowNotify,
+	callflag.States,
+	callflag.ReadOnly,
+	callflag.NoneFlag,
+}
+
+var generateWrapperCmd = cli.Command{
+	Name:        "generate-wrapper",
+	Usage:       "generate wrapper to use in other contracts",
+	UsageText:   "neo-go contract generate-wrapper --manifest manifest.json --out file.go",
+	Description: ``,
+	Action:      contractGenerateWrapper,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "package, p",
+			Usage: "Package to use for Go code (default: contract name)",
+		},
+		cli.StringFlag{
+			Name:  "config, c",
+			Usage: "Configuration file to use",
+		},
+		cli.StringFlag{
+			Name:  "manifest, m",
+			Usage: "Read contract manifest (*.manifest.json) file",
+		},
+		cli.StringFlag{
+			Name:  "out, o",
+			Usage: "Output of the compiled contract",
+		},
+		cli.StringFlag{
+			Name:  "hash",
+			Usage: "Smart-contract hash",
+		},
+		cli.StringSliceFlag{
+			Name: "override",
+			Usage: `Override type for methods. Should have form <method>.<name>=<type>.
+Type should be fully-qualified name of the type. For return use <method>=<type>'.
+Common packages from interop can be aliases with "storage", "iterator" or "ledger".`,
+		},
+		cli.StringSliceFlag{
+			Name: "callflag",
+			Usage: `Override call flags. Should have form <method>=<callflag>.
+Call flag can be one of All, ReadStates, WriteStates, AllowCall, AllowNotify, States, ReadOnly, NoneFlag.`,
+		},
+	},
+}
+
+// contractGenerateWrapper deploys contract.
+func contractGenerateWrapper(ctx *cli.Context) error {
+	m, _, err := readManifest(ctx.String("manifest"))
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("can't read contract manifest: %w", err), 1)
+	}
+
+	cfg := binding.NewConfig()
+	if cfgPath := ctx.String("config"); cfgPath != "" {
+		bs, err := ioutil.ReadFile(cfgPath)
+		if err != nil {
+			return cli.NewExitError(fmt.Errorf("can't read config file: %w", err), 1)
+		}
+		err = yaml.Unmarshal(bs, &cfg)
+		if err != nil {
+			return cli.NewExitError(fmt.Errorf("can't parse config file: %w", err), 1)
+		}
+	}
+
+	if pkgName := ctx.String("package"); pkgName != "" {
+		cfg.Package = pkgName
+	}
+	cfg.Manifest = m
+
+	if ctx.String("hash") != "" {
+		h, err := util.Uint160DecodeStringLE(ctx.String("hash"))
+		if err != nil {
+			return cli.NewExitError(fmt.Errorf("invalid contract hash: %w", err), 1)
+		}
+		cfg.Hash = h
+	}
+
+	err = parseOverrides(ctx, &cfg)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	err = parseCallFlags(ctx, &cfg)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	f, err := os.Create(ctx.String("out"))
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("can't create output file: %w", err), 1)
+	}
+	defer f.Close()
+
+	cfg.Output = f
+
+	err = binding.Generate(cfg)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during generation: %w", err), 1)
+	}
+	return nil
+}
+
+func parseCallFlags(ctx *cli.Context, cfg *binding.Config) error {
```

Comment:
Why do we still have this? I thought config file solves the problem.

Reply:
IIUC, now we have the ability to specify them both via config and via argument. So maybe we can keep both options? For those who don't want to create the configuration file for a single binding generation.

Reply:
Exactly, this allows to extend everything via CLI arguments. Though after #2359 this may not be needed at all.

Reply:
Well, this looks like additional code to maintain to me. Does it add any value? I'm not sure, configuration solves the same problem.